core/object/script_language_extension.h

```diff
@@ -663,6 +663,14 @@ class ScriptInstanceExtension : public ScriptInstance {
 		if (native_info->get_property_list_func) {
 			uint32_t pcount;
 			const GDNativePropertyInfo *pinfo = native_info->get_property_list_func(instance, &pcount);
+
+#ifdef TOOLS_ENABLED
+			Ref<Script> script = get_script();
+			if (script->is_valid() && pcount > 0) {
```

Comment:
Hi @object71 do you remember wy you use `x->valid()` in stead off `x.valid()`?
-> makes my GDExtension crash so before making a PR I want to check if it was intentional.

Reply:
I think it might need both (`Ref::is_valid` and `Script::is_valid`)

Reply:
It indeed should be `script.is_valid()` as it's the reference which needs to be validated before dereferencing it down below in `script->get_class_category()`.

---

 `script.is_valid()` will call:
https://github.com/godotengine/godot/blob/a83eb16fba5bb3da086b41cbd79e6f95b09eb8ee/core/object/ref_counted.h#L201

---

 `script->is_valid()` will dereference the stored `reference` pointer (possibly `nullptr`):
https://github.com/godotengine/godot/blob/a83eb16fba5bb3da086b41cbd79e6f95b09eb8ee/core/object/ref_counted.h#L104-L106
and call:
 https://github.com/godotengine/godot/blob/a83eb16fba5bb3da086b41cbd79e6f95b09eb8ee/core/object/script_language.h#L150

Reply:
What do you mean by "possibly crashing" ?

Reply:
> What do you mean by "possibly crashing" ?

@Gallilus I mean if the `Ref<Script> script`'s private`Script *reference` member is indeed `nullptr` then `script->is_valid()` is equivalent to `nullptr->is_valid()`. And dereferencing `nullptr` is an undefined behavior, might result in a crash.

Hence before any `script->...` calls it should be ensured (e.g. by calling `script.is_valid()`) that it can be dereferenced.

Reply:
This may sound silly but I'm not jet a regular in C++ conversations.
With dereference you mean going true the pointer in the object to do stuff.
Not deleting it from memory.

Reply:
Also, I had a .ptr() in my code so ...

```
    MyClass my_object = memnew(MyClass);
    Ref<MyClass > reference_1 = *my_object;
    Ref<MyClass > reference_2a = reference_1;
    Ref<MyClass > reference_2b = reference_1.ptr();
    if (refrence_2b.is_valid()){
        // Is reference_2 a or b valid?
        // Will reference.is_valid() && refrence->is_valid() catch this problem?
    }
```